### DISCUSSION TITLE
How much memory allocated to OPA server.

### DISCUSSION THREAD

**Discussion by Akash-Adkute** [open-policy-agent/community#390] [OPA and Rego]
Hi,
I'm beginner to OPA and wnated to know memory management in OPA,
1.How much memory allocated to opa server by default to store data and policy(can we specify memory size)?
2.Is there any way we can allocate external memory to opa server,if yes then will it affect on performance?

**Comment by anderseknert** [comment]
Hi there! 

OPA will use as much memory as required, and how much that is depends on the number of policies, and the size of the data you have loaded into OPA. 

Setting limits for what resources a process may allocate is commonly done via the environment, whether it's directly in the OS, or through container orchestration tools like Docker / Kubernetes.
